### PR TITLE
[Security] Make Login Rate Limiter also case insensitive for non-ascii user identifiers

### PR DESCRIPTION
**Pull request by Seldaek**
| Q             | A
| ------------- | ---
| Branch?       | 5.2
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

As per discussion in https://github.com/symfony/symfony/pull/41156

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ protected function getLimiters(Request $request): array
     {
         return [
             $this->globalFactory->create($request->getClientIp()),
-            $this->localFactory->create(strtolower($request->attributes->get(Security::LAST_USERNAME)).'-'.$request->getClientIp()),
+            $this->localFactory->create(mb_strtolower($request->attributes->get(Security::LAST_USERNAME)).'-'.$request->getClientIp()),
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php]: we should not rely on the global php.ini config
this should instead be:
`\function_exists('mb_strtolower') && preg_match('//u', $v) ? mb_strtolower($v, 'UTF-8') : strtolower($v);`

Comment (derrabus) [reply src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php]: Do we need `function_exists('mb_strtolower')`? The polyfill should make sure that the method exists, shouldn't it?

Comment (Seldaek) [reply src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php]: @derrabus that's my understanding too.

@nicolas-grekas are you sure about passing `'UTF-8'`? I don't mind changing, but seeing as utf-8 has been the default for a while, and if someone explicitly sets it to something else I'd assume they have a good reason and maybe it makes sense to follow their setting.

Comment (nicolas-grekas) [reply src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php]: true, about `function_exists`
for the ini setting, I think the code base is decoupled from it and we always use that kind of calls.
no global state rulez :)

Comment (Seldaek) [reply src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php]: Ok done.. I kept the preg_match() guard, not sure if it's really needed or not? Is the polyfill not enough?

**Review by chalasr — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php]: It's needed yes :)

**Comment by derrabus** [comment]
Thank you Jordi.

**Pull request merged**